ndcube/meta.py

```diff
@@ -0,0 +1,374 @@
+        if isinstance(axis, numbers.Integral):
+            axis = (axis,)
+        if len(axis) == 0:
+            return ValueError(axis_err_msg)
+        if self.shape is None:
+            raise TypeError("Meta instance does not have a shape so new metadata "
+                            "cannot be assigned to an axis.")
+        # Verify each entry in axes is an iterable of ints or a scalar.
+        if not (isinstance(axis, collections.abc.Iterable) and all([isinstance(i, numbers.Integral)
+                                                                    for i in axis])):
+            return ValueError(axis_err_msg)
+        axis = np.asarray(axis)
+        if _not_scalar(value):
+            axis_shape = tuple(self.shape[axis])
+            if not _is_grid_aligned(value, axis_shape) and not _is_axis_aligned(value, axis_shape):
+                raise ValueError(
+                    f"{key} must have shape {tuple(self.shape[axis])} "
+                    f"as its associated axes {axis}, ",
+                    f"or same length as number of associated axes ({len(axis)}). "
+                    f"Has shape {value.shape if hasattr(value, 'shape') else len(value)}")
+        return axis
+
+    @property
+    def comments(self):
+        return self._comments
+
+    @property
+    def axes(self):
+        return self._axes
+
+    @property
+    def shape(self):
+        return self._data_shape
+
+    def add(self, name, value, comment=None, axis=None, overwrite=False):
+        """
+        Add a new piece of metadata to instance.
+
+        Parameters
+        ----------
+        name: `str`
+            The name/label of the metadata.
+
+        value: Any
+            The value of the metadata. If axes input is not None, this must have the
+            same length/shape as those axes as defined by ``self.shape``.
+
+        comment: `str` or `None`
+            Any comment associated with this metadata. Set to None if no comment desired.
+
+        axis: `int`, iterable of `int`, or `None`
+            The axis/axes with which the metadata is linked. If not associated with any
+            axis, set this to None.
+
+        overwrite: `bool`, optional
+            If True, overwrites the entry of the name name if already present.
+        """
+        if name in self.keys() and overwrite is not True:
+            raise KeyError(f"'{name}' already exists. "
+                           "To update an existing metadata entry set overwrite=True.")
+        if comment is not None:
+            self._comments[name] = comment
+        if axis is not None:
+            axis = self._sanitize_axis_value(axis, value, name)
+            self._axes[name] = axis
+        elif name in self._axes:
+            del self._axes[name]
+        # This must be done after updating self._axes otherwise it may error.
+        self.__setitem__(name, value)
+
+    def remove(self, name):
```

Comment:
I think you want to maybe make this `__delitem__` so `del mymeta[name]` works. I don't see any particular reason why we need `.remove` as well.